More towers 2

vincetiu8 opened the pull request:
Adds the 'm_cell', a (slightly overpowered) tower that shoots exploding bullets slowly that damage enemies. This PR also fixes bullet tracking issues.

vincetiu8 committed:
Add the m_cell into the game

This adds the m_cell, a slow-firing tower that does lots of damage to enemies.

vincetiu8 committed:
Fix bullet shooting

This fixes bullet firing, tracking and movement, which were all glitchy because I don't know how to do basic trigonometry.

vincetiu8 committed:
Fixes tower gun images

This edits the tower gun images so they are facing upwards and have the least amount of transparent space as needed, to make image rotation faster.

vincetiu8 committed:
Add explosive bullets

This adds explosive bullets to the m_cell, which admittedly does make it overpowered but is a nice feature we can play around with.

Graphcalibur reviewed (changes requested):
LGTM but the t_cell image on the TowerSelectMenu and UI looks strange.

vincetiu8 committed:
Fix images

This converts the images from index back to RGB. Note this gives LIBPNG errors, but I don't have a good way of fixing them right now.

Graphcalibur reviewed (approved):
LGTM!
